docs/user-guide/creating-application/eso/hashicorp-eso.md

```diff
@@ -0,0 +1,59 @@
+# HashiCorp Vault
+
+To incorporate secrets from **HashiCorp Vault**, you need to create a generic Kubernetes secret that will be used for vault authentication. This involves creating a Kubernetes secret in the specific namespace where your application will be deployed. 
+The secret should store the base64-encoded password or token obtained from vault. To simplify the process, you can utilize the Devtron generic chart. An example yaml is given below:
+
+```yaml
+apiVersion: v1
+kind: Secret
+type: Opaque
+data:
+   token: <vault-password>
+metadata:
+   name: vault-token
+   namespace: <namespace>
+```
+
+**Note:** Please note that you don't need to create the Kubernetes secret every time you create an External Secret for the corresponding namespace.
+
+Once you have created the generic secret, follow these steps in the application's Secrets section:
+
+**1. Create a new secret.**
+
+To add a new secret to the application, go to the `App Configuration` section of the application. Then, navigate to the left pane and select the `Secrets` option and click on the `Add Secret` button.
```

Comment:
To add a new secret to the application, go to the `App Configuration` section of the application. Then, navigate to the left pane and select the `Secrets` option and click the **Add Secret** button.